src/main/java/com/exasol/projectkeeper/validators/DeletedFilesValidator.java

```diff
@@ -38,19 +38,21 @@ public DeletedFilesValidator(final Path projectDirectory, final ExcludedFilesMat
     }
 
     @Override
-    public Validator validate(final Consumer<ValidationFinding> findingConsumer) {
-        FILES_THAT_MUST_NOT_EXIST.forEach((fileThatMustNotExist, reason) -> {
+    public List<ValidationFinding> validate() {
+        return FILES_THAT_MUST_NOT_EXIST.entrySet().stream().map((entry) -> {
+            final var fileThatMustNotExist = entry.getKey();
+            final var reason = entry.getValue();
             final var pathThatMustExist = this.projectDirectory.resolve(fileThatMustNotExist);
             if (this.excludedFiles.isFileExcluded(this.projectDirectory.relativize(pathThatMustExist))) {
```

Comment:
What do you check here? I think extracting a method with a clear name would help a lot to understand the code

Reply:
I check if the file is excluded. 
Not sure how to improve readability. The method is already called `isFileExcluded`